components/camel-aws-sns/src/main/java/org/apache/camel/component/aws/sns/SnsProducer.java

```diff
@@ -104,6 +106,13 @@ private String determineMessageStructure(Exchange exchange) {
                     mav.setDataType("String");
                     mav.withStringValue(value.toString());
                     result.put(entry.getKey(), mav);
+                } else if (value instanceof List && ((List) value).size()>0) {
+                    // Avoiding reliance on .toString()
+                    String delimiter = "\", \"", prefix = "[\"", suffix = "\"]";
```

Comment:
This can be on multiple lines, for clarity

Reply:
done